### PR TITLE
[ReleasePR Microsoft.Azure.Management.Automanage] Microsoft.Automanage SDK for 2020 06 30 preview 

### PR DESCRIPTION
**Pull request by stowen-msft**
Includes the SDK Release for Microsoft.Automanage, tested and validated client.  Inlcudes recorded results, works great.

**Target release date 09/21/2020**

OpenAPI Spec already completed here : https://github.com/Azure/azure-rest-api-specs/pull/10344

### CODE REVIEW

**Review by allenjzhang — CHANGES_REQUESTED** [review]
I am adding CI mgmt pipeline update for you. In the meantime, could you:

 - Add a link to the openapi spec review PR in description

 - Check in the metadata TXT file generated when you run `generate.ps1`[(details)](https://github.com/Azure/adx-documentation-pr/blob/master/engineering/adx_netsdk_process.md#sdk-generation-from-updated-spec)

 - Update versions in csproj and **`assemblyinfo.cs`** (missing) if you are going to release

 - I don't believe you need any Package reference in src\.csproj. Please remove and compile again to verify.
 
-  Assembly/Package name casing is off. The proper casing is Microsoft.Azure.Management.Auto**M**anagement. This should be corrected everywhere.

Thanks

**Comment by stowen-msft** [comment]
Swagger / OpenAPI link - https://github.com/Azure/azure-rest-api-specs/pull/10344

**Comment by stowen-msft** [comment]
Hi Allen, I cannot get `generate.ps1` to work, so I generated my code directly from npm using AutoRest...

I'm not sure where / how to get a metadata.txt file in this case (I don't have the cmdlet needed, `Start-AutoRestGeneration`)

**Comment by stowen-msft** [comment]
✔Add a link to the openapi spec review PR in description

😮Check in the metadata TXT file generated when you run generate.ps1(details)

✔Update versions in csproj and assemblyinfo.cs (missing) if you are going to release

✔I don't believe you need any Package reference in src.csproj. Please remove and compile again to verify.

✔Assembly/Package name casing is off. The proper casing is Microsoft.Azure.Management.AutoManagement. This should be corrected everywhere.

**Comment by m-nash** [comment]
I believe the casing needs to get updated everywhere including *.csproj file name, *.sln file name and directory name
Still not seeing the metadata.txt file

**Comment by stowen-msft** [comment]
Metadata file was added and all casing fixed. Anything else we need for this one?

Trying to get it in for an ignite week release.

Sent from Outlook Mobile<https://aka.ms/blhgte>

________________________________

**Comment by stowen-msft** [comment]
Thanks @allenjzhang , tried merging with upstream as I saw some PRs that looked relevant for fixing build issues, but I am still failing with in the build pipelines

```
 "D:\a\1\s\mgmt.proj" (RunTests target) (1) ->
                   "D:\a\1\s\sdk\automanage\Microsoft.Azure.Management.AutoManage\tests\Microsoft.Azure.Management.AutoManage.Tests.csproj" (default target) (140:10) ->
                   (ResolvePackageAssets target) -> 
                     C:\Program Files\dotnet\sdk\3.1.302\Sdks\Microsoft.NET.Sdk\targets\Microsoft.PackageDependencyResolution.targets(234,5): error NETSDK1005: Assets file 'D:\a\1\s\artifacts\obj\Microsoft.Azure.Management.AutoManage.Tests\project.assets.json' doesn't have a target for '.NETCoreApp,Version=v3.1'. Ensure that restore has run and that you have included 'netcoreapp3.1' in the TargetFrameworks for your project. [D:\a\1\s\sdk\automanage\Microsoft.Azure.Management.AutoManage\tests\Microsoft.Azure.Management.AutoManage.Tests.csproj]
```

File: sdk/automanage/Microsoft.Azure.Management.AutoManage/src/Microsoft.Azure.Management.AutoManage.csproj [hunk 1]
```diff
@@ -0,0 +1,18 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+  <!-- Please do not move/edit code below this line -->
+  <Import Project="$(RepoEngPath)/mgmt/AzSdk.reference.props" />
+  <!-- Please do not move/edit code below this line -->
+  
+  <PropertyGroup>
+    <TargetFramework>netstandard2.0</TargetFramework>
```
Comment (allenjzhang) [inline sdk/automanage/Microsoft.Azure.Management.AutoManage/src/Microsoft.Azure.Management.AutoManage.csproj:7]: I think the errors are related to the TargetFramework being explicitly coded here. I also notice that you are missing related packageid tags etc. Please reference existing .csproj for other RPs.

**Review by allenjzhang — CHANGES_REQUESTED** [review]

File: sdk/automanage/Microsoft.Azure.Management.AutoManage/src/Microsoft.Azure.Management.AutoManage.csproj (hunk 1 above)
Comment (stowen-msft) [reply sdk/automanage/Microsoft.Azure.Management.AutoManage/src/Microsoft.Azure.Management.AutoManage.csproj]: thank you very much, done.

File: eng/mgmt/mgmtmetadata/automation_resource-manager.txt [hunk 2]
```diff
@@ -0,0 +1,14 @@
+﻿Installing AutoRest version: v2
```
Comment (allenjzhang) [inline eng/mgmt/mgmtmetadata/automation_resource-manager.txt:1]: this file is automation_resource-manager.txt indicating it was not from running the generate.ps1. That file needs to be corrected. This file should have been **automanage**_resource-manager.txt

File: sdk/automanage/Microsoft.Azure.Management.AutoManage/src/generate.ps1 [hunk 3]
```diff
@@ -0,0 +1 @@
+Start-AutoRestCodeGeneration -ResourceProvider "automation/resource-manager" -AutoRestVersion "v2"
```
Comment (allenjzhang) [inline sdk/automanage/Microsoft.Azure.Management.AutoManage/src/generate.ps1:1]: Should be **automanage**/resource-manager

File: sdk/automanage/Microsoft.Azure.Management.AutoManage/src/AzSdk.RP.props [hunk 4]
```diff
@@ -0,0 +1,7 @@
+﻿<Project ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <!--This file and it's contents are updated at build time moving or editing might result in build failure. Take due deligence while editing this file-->
+  <PropertyGroup>
+    <AzureApiTag>Automation_2015-10-31;Automation_2018-01-15;Automation_2017-05-15-preview;Automation_2018-06-30;</AzureApiTag>
```
Comment (allenjzhang) [inline sdk/automanage/Microsoft.Azure.Management.AutoManage/src/AzSdk.RP.props:4]: This should be your automanage api-version.

File: sdk/automanage/Microsoft.Azure.Management.AutoManage/src/CHANGELOG.md [hunk 5]
```diff
@@ -0,0 +1,11 @@
+﻿# Microsoft.Azure.Management.Automanage release notes
+
+## Changes in 0.6.0-preview
```
Comment (allenjzhang) [inline sdk/automanage/Microsoft.Azure.Management.AutoManage/src/CHANGELOG.md:3]: Can you correct the version here?

**Review by allenjzhang — CHANGES_REQUESTED** [review]

File: sdk/automanage/Microsoft.Azure.Management.AutoManage/src/AzSdk.RP.props (hunk 4 above)
Comment (stowen-msft) [reply sdk/automanage/Microsoft.Azure.Management.AutoManage/src/AzSdk.RP.props]: Good catch, fixed!

File: eng/mgmt/mgmtmetadata/automation_resource-manager.txt (hunk 2 above)
Comment (stowen-msft) [reply eng/mgmt/mgmtmetadata/automation_resource-manager.txt:1]: Reran generate and updated, thank you.

File: sdk/automanage/Microsoft.Azure.Management.AutoManage/tests/Microsoft.Azure.Management.AutoManage.Tests.csproj [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+  <Import Project="$(RepoEngPath)/mgmt/AzSdk.test.reference.props" />
+  <PropertyGroup>
+    <PackageId>AutoManage.Tests</PackageId>
+    <Description>AutoManage.Tests Class Library</Description>
+    <AssemblyName>AutoManage.Tests</AssemblyName>
+    <VersionPrefix>0.1.0-preview</VersionPrefix>
+    <RootNamespace>Microsoft.Azure.Management.AutoManage.Tests</RootNamespace>
+  </PropertyGroup>
+  <ItemGroup>
+    <None Update="SessionRecords\OperationsTests\ListOperationsTest.json">
```
Comment (allenjzhang) [inline sdk/automanage/Microsoft.Azure.Management.AutoManage/tests/Microsoft.Azure.Management.AutoManage.Tests.csproj:11]: Test failure caused by not finding SessionRecords. Please replace with following to copy over the **all** of the files
`  <ItemGroup>
    <None Update="SessionRecords\**\*.json">
      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
    </None>
  </ItemGroup>`

**Review by allenjzhang — CHANGES_REQUESTED** [review]

File: sdk/automanage/Microsoft.Azure.Management.AutoManage/tests/Microsoft.Azure.Management.AutoManage.Tests.csproj (hunk 6 above)
Comment (stowen-msft) [reply sdk/automanage/Microsoft.Azure.Management.AutoManage/tests/Microsoft.Azure.Management.AutoManage.Tests.csproj]: thank you Allen, fixed.

File: sdk/automanage/Microsoft.Azure.Management.AutoManage/tests/Microsoft.Azure.Management.AutoManage.Tests.csproj [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
+    </None>
+  </ItemGroup>
+  
+  <ItemGroup>
+    <PackageReference Include="Microsoft.Azure.Management.ResourceManager" Version="1.6.0-preview" />    
+    <ProjectReference Include="..\src\Microsoft.Azure.Management.AutoManage.csproj" />
+  </ItemGroup>
+  
+  <ItemGroup>
```
[the remark this answers is not in the document]
Comment (stowen-msft) [reply sdk/automanage/Microsoft.Azure.Management.AutoManage/tests/Microsoft.Azure.Management.AutoManage.Tests.csproj]: Fixed!

**Review by allenjzhang — APPROVED** [review]

**Comment by mromanch** [comment]
> * Assembly/Package name casing is off. The proper casing is Microsoft.Azure.Management.Auto**M**anagement. This should be corrected everywhere.

@allenjzhang -- The name of our service is Azure Automanage. We had this name reviewed by LT, marketing etc... The name and the case are based on the documentation in the Microsoft Cloud Style Guide.

To be clear, _automanage_ is a single word, so it is capitalized as **Automanage**. This is compared to something like virtual machine, capitalized as VirtualMachine or guest configuration, capitalized as GuestConfiguration.

This PR has completed, and it is my understanding that we are already past the window for any changes that can go into the soonest build. This means that our product is forever stuck with this badly capitalized name.

Is there any way we can correct the name and still get in the next SDK release, i.e. the release that will be available in time for Ignite?

**Pull request merged**